src/data/mv_polynomial.lean

```diff
@@ -890,6 +965,86 @@ begin
   rwa [←to_finset_to_multiset, multiset.mem_to_finset] at h,
 end
 
+lemma vars_add_subset (p q : mv_polynomial σ α) :
+  (p + q).vars ⊆ p.vars ∪ q.vars :=
+begin
+  intros x hx,
+  simp only [vars, finset.mem_union, multiset.mem_to_finset] at hx ⊢,
+  simpa using multiset.mem_of_le (degrees_add _ _) hx,
+end
+
+lemma vars_add_of_disjoint (h : disjoint p.vars q.vars) :
+  (p + q).vars = p.vars ∪ q.vars :=
+begin
+  apply finset.subset.antisymm (vars_add_subset p q),
+  intros x hx,
+  simp only [vars, multiset.disjoint_to_finset] at h hx ⊢,
+  rw [degrees_add_of_disjoint h, multiset.to_finset_union],
+  exact hx
+end
+
+section sum
+
+variables {ι : Type*} (t : finset ι) (φ : ι → mv_polynomial σ α)
+
+lemma vars_sum_subset :
+  (∑ i in t, φ i).vars ⊆ finset.bind t (λ i, (φ i).vars) :=
+begin
+  apply t.induction_on,
+  { simp },
+  { intros a s has hsum,
+    rw [finset.bind_insert, finset.sum_insert has],
+    refine finset.subset.trans (vars_add_subset _ _)
+      (finset.union_subset_union (finset.subset.refl _) _),
+    assumption }
+end
+
+lemma vars_sum_of_disjoint (h : pairwise $ disjoint on (λ i, (φ i).vars)) :
+  (∑ i in t, φ i).vars = finset.bind t (λ i, (φ i).vars) :=
+begin
+  apply t.induction_on,
+  { simp },
+  { intros a s has hsum,
+    rw [finset.bind_insert, finset.sum_insert has, vars_add_of_disjoint, hsum],
+    unfold pairwise on_fun at h,
+    rw hsum,
+    simp only [finset.disjoint_iff_ne] at h ⊢,
+    intros v hv v2 hv2,
+    rw finset.mem_bind at hv2,
+    rcases hv2 with ⟨i, his, hi⟩,
+    refine h a i _ _ hv _ hi,
+    rintro rfl,
+    contradiction }
+end
+
+end sum
+
+section map
+
+variables [bcsr : comm_semiring β] (f : α →+* β)
```

Comment:
Is this named `bcsr` for a reason?

Reply:
Nope, I think it got copied from a section where it had to be `omit`ted later.